crates/turborepo-lib/src/run/mod.rs

```diff
@@ -397,6 +413,10 @@ impl<'a> Run<'a> {
             )
             .await?;
 
+        if let Some(analytics_handle) = analytics_handle {
+            analytics_handle.close_with_timeout().await;
```

Comment:
We need to flush/close the analytics regardless of how the function finishes, not just if we hit this line.